expan/core/experiment.py

```diff
@@ -339,3 +374,19 @@ def run_goodness_of_fit_test(self, observed_freqs, expected_freqs, alpha=0.01, m
             raise ValueError("Variant split check was cancelled since observed or expected frequencies "
                              "are less than 2.")
         return split_is_unbiased, p_value
+
+def _choose_threshold_type(data):
+    """ Heuristics used to decide what filtering method to use."""
+    assert len(data), 'data should be non-empty'
+    assert data.min,  'data should have min method'
+    assert data.max,  'data should have max method'
+    min, max = data.min(), data.max()
+
+    if min < 0.0 and max > 0.0:
+        return 'two-sided'
+    elif max > 0.0:
+        return 'upper'
+    elif min < 0.0:
+        return 'lower'
+    else:
+        raise ValueError("Both min and max of data are zero.")
```

Comment:
We shouldn't raise an error here. We could quite legitimately have vectors full of zeroes, especially for small datasets

In general, we should treat `0` (and `np.inf`, `-np.inf` and `np.nan`) as legitimate values. Our derived KPIs will often be `0/0 = np.nan`

How do we expect outlier filtering to interact with `np.nan`? If we have 1'000'000 values, of which 300'000 are `np.nan`, and we have 99% as our threshold, will we remove 10'000 or 3'000?

Reply:
I'm returning 'two-sided' here now, too.

Comparison of a nan and any normal number always returns `False`, so they're never taken out.